packages/next/src/server/render.tsx

```diff
@@ -1057,7 +1057,8 @@ export async function renderToHTML(
 
   // Avoid rendering page un-necessarily for getServerSideProps data request
   // and getServerSideProps/getStaticProps redirects
-  if ((isDataReq && !isSSG) || (renderOpts as any).isRedirect) {
+  // In development the incrementalCache is not used so we short-circuit bypassing the html rendering.
+  if ((isDataReq && (!isSSG || dev)) || (renderOpts as any).isRedirect) {
```

Comment:
Hmm does the benefit of avoiding rendering here in dev offset the potential downside that a rendering error isn't caught?

Reply:
It's specifically needed as Turbopack handles data requests differently in that the team implemented a reverse of the getStaticProps/getServerSideProps tree shaking transform to allow better HMR of getStaticProps/getServerSideProps changes. So you end up with a file that doesn't have the component code, only the code used in getStaticProps.

Reply:
Sounds like that will need to be changed soon as we're changing how turbopack is handled and wouldn't work in production.